dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/AuthorizationRestRepository.java

```diff
@@ -0,0 +1,268 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.dspace.app.rest.Parameter;
+import org.dspace.app.rest.SearchRestMethod;
+import org.dspace.app.rest.authorization.Authorization;
+import org.dspace.app.rest.authorization.AuthorizationFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeatureService;
+import org.dspace.app.rest.authorization.AuthorizationRestUtil;
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.model.AuthorizationRest;
+import org.dspace.app.rest.model.BaseObjectRest;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.service.EPersonService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+
+/**
+ * This is the repository responsible to manage Authorization Rest object
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ */
+
+@Component(AuthorizationRest.CATEGORY + "." + AuthorizationRest.NAME)
+public class AuthorizationRestRepository extends DSpaceRestRepository<AuthorizationRest, String> {
+
+    private static final Logger log = LoggerFactory.getLogger(AuthorizationRestRepository.class);
+
+    @Autowired
+    private AuthorizationFeatureService authorizationFeatureService;
+
+    @Autowired
+    private AuthorizationRestUtil authorizationRestUtil;
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    @Autowired
+    private EPersonService epersonService;
+
+    @Autowired
+    protected ConverterService converter;
+
+    @Override
+    @PreAuthorize("hasPermission(#id, 'authorization', 'READ')")
```

Comment:
Does this imply you can only request the "features" of resources you have read rights for?
I was under the assumption this could be used to identify whether the current user has permissions to download a bitstream on the item which are displayed in DSpace 6 with a lock icon

Reply:
Yes. We don't have enough information at this point to make a different decision. Limit access to the authorization object that indeed include the scope object to people that can READ the scope object make sense and is safe. For bitstream that you cannot completely READ (you can know that they exists but you cannot download them) we should find a suitable solution.
Essentially we will need a way to differentiate between READ only metadata and READ content, the easier would probably be use READ for just metadata relaxing the permission evaluator for bitstreams to always trust user that have READ on the Item for "normal bundle" and introduce a DOWNLOAD permission that will really use the READ resourcepolicy on the bitstream. Again I recommend to put this discussion and the consequent implementation in a separate task

Reply:
Could you in that case provide feedback in https://jira.lyrasis.org/browse/DS-4454 on how you'd expect it to be possible to identify whether the current user can download the bitstream?

Reply:
https://jira.lyrasis.org/browse/DS-4454?focusedCommentId=64231&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-64231